SourceKitLSPDevUtils/Sources/ConfigSchemaGen/JSONSchema.swift

```diff
@@ -126,13 +135,53 @@ struct JSONSchemaBuilder {
       schema.properties = properties
       schema.required = required
     case .enum(let enumInfo):
-      schema.type = "string"
-      schema.enum = enumInfo.cases.map(\.name)
-      // Set `markdownEnumDescriptions` for better rendering in VSCode rich hover
-      // Unlike `description`, `enumDescriptions` field is not a part of JSON Schema spec,
-      // so we only set `markdownEnumDescriptions` here.
-      if enumInfo.cases.contains(where: { $0.description != nil }) {
-        schema.markdownEnumDescriptions = enumInfo.cases.map { $0.description ?? "" }
+      let hasAssociatedTypes = enumInfo.cases.contains { $0.associatedProperties != nil && !$0.associatedProperties!.isEmpty }
```

Comment:
I personally think it reads nicer like this because you can just infer that `nil` is considered empty. But I know that some people disagree.
 
```suggestion
      let hasAssociatedTypes = enumInfo.cases.contains { !($0.associatedProperties?.isEmpty ?? true) }
```